app/server/utils/ussd/kenya_ussd_processor.py

```diff
@@ -76,5 +76,19 @@ def replace_vars(menu: UssdMenu, ussd_session: UssdSession, display_text: str, u
                     replacements.append(['%remaining_attempts%', "You have {} attempts remaining.".format(3 - user.pin_failed_attempts)])
             else:
                 replacements.append(['%remaining_attempts%', ''])
+        elif 'directory_listing' == menu.name or 'send_token_reason' == menu.name:
+            usages = user.get_most_relevant_transfer_usage()
+            menu_options = ''
+            for i, usage in enumerate(usages):
+                business_usage_string = None
+                if usage.translations is not None and user.preferred_language is not None:
+                    business_usage_string = usage.translations.get(
+                        user.preferred_language)
+                    print(business_usage_string)
+                if business_usage_string is None:
+                    business_usage_string = usage.name
+                message_option = '%d. %s ' % (i+1, business_usage_string)
+                menu_options += message_option
+            replacements.append(['%options%', menu_options])
```

Comment:
Is there a test that hits this code?
I was trying work out how the USSD menu option number for a custom increments with respect to the hard-coded options, but i couldn't find any way to this code...

Reply:
+1 to this. also, how does the dynamic options get hooked up to actual usage? for example, if a user responds with 2 for the directory listing, how does the state machine know what "2" corresponds to in terms of transfer usages? i didn't see that bit. an edge case to consider would be if their `get_most_relevant_transfer_usage` changes in between displaying it to customer and them responding

Reply:
Ah thanks for the explanation I did not think that through yet.

Reply:
What should directory listening return? For example if you select option 1. Food where should you go? Whould you get back list of all user that fall into that business category? Or the transfer accounts related to that user?

Reply:
Would it happen that the `get_most_relevant_transfer_usage` changes in between? Since this is based on the transfers that a user has sent. So they themselves would have to send transfers while the menu is showing to change that ranking. Or are the multiple people accessing one user account? Or am i missing something else?

Reply:
> What should directory listening return? For example if you select option 1. Food where should you go?
per comment on slack, don't worry about that for this ticket, if you can just have it translate to the transfer usage, we'll address that in the other ticket
> Or are the multiple people accessing one user account? 
multiple users may share one group account so this is a potential edge case that could happen

Reply:
> @RubenGeo just pinging you this last one incase the comment slipped through?

Which one?

> @enjeyw we have defaults that are displayed if there's not enough to fill the list. #2 is a great question, since our "other" behavior is now different than what's on GE. for transfer reason, "other" will always be the last option and choosing that takes them to the option to type in a new transfer reason (@RubenGeo don't know if you did it this way already, i think i left that out in the specs), which would have the same behavior as our admin interface of `find_or_create` transfer usage. for directory listing, we should talk to will. "other" could just show every other kind not listed in the top 10...?

So should I revert the "other" back to show other categories instead of a seperate option? And then create a menu that show all options that were not in the top 10? Does it not make more sense to have like 10. 'Show other options' and translate that to sw?

Reply:
Wait so when a user sends tokens to another user and they can't find their intended transfer usage in orginal list of transfer usage. They're allowed to add a new tranfer usage. 

I thought creating new transer usage was for admin? And the USSD interface is not really handy for adding new transfer usage as you would also need to set translations?